Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/Systems/AnalyticsEmotesSystem.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.AvatarRendering.AvatarShape;
+using DCL.AvatarRendering.Emotes;
+using DCL.Diagnostics;
+using DCL.PerformanceAndDiagnostics.Analytics;
+using ECS;
+using ECS.Abstract;
+using Segment.Serialization;
+
+namespace DCL.Analytics.Systems
+{
+    [LogCategory(ReportCategory.ANALYTICS)]
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(AvatarGroup))]
+    [UpdateBefore(typeof(CharacterEmoteSystem))]
+    public partial class AnalyticsEmotesSystem : BaseUnityLoopSystem
+    {
+        private readonly IAnalyticsController analytics;
+
+        private readonly IRealmData realmData;
+        private readonly Entity playerEntity;
+
+        private string lastEmoteId = string.Empty;
+
+        public AnalyticsEmotesSystem(World world, IAnalyticsController analytics, IRealmData realmData, in Entity playerEntity) : base(world)
+        {
+            this.analytics = analytics;
+            this.realmData = realmData;
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void Update(float t)
+        {
+            if (!realmData.Configured) return;
+
+            if (World.TryGet(playerEntity, typeof(CharacterEmoteIntent), out object? intent))
```

Comment:
You could try declaring the specific type and avoid casting: `World.TryGet(playerEntity, out CharacterEmoteIntent intent)`

Reply:
nice, thank you. I was looking for this API, but didn't found it neither from Rider or from Arch docs. So I was thinking that it doesn't exist. ChatGpt helped this time ;) 
`World.TryGet<CharacterEmoteIntent>(playerEntity, out var emoteIntent)`